external-crates/move/move-compiler/src/diagnostics/mod.rs

```diff
@@ -340,6 +362,69 @@ macro_rules! diag {
     }};
 }
 
+impl WarningFilters {
+    pub fn new() -> Self {
+        Self::Empty
+    }
+
+    pub fn is_filtered(&self, diag: &Diagnostic) -> bool {
+        self.is_filtered_by_info(&diag.info)
+    }
+
+    fn is_filtered_by_info(&self, info: &DiagnosticInfo) -> bool {
+        match self {
+            WarningFilters::All => info.severity() == Severity::Warning,
+            WarningFilters::Specified { category, codes } => {
+                info.severity() == Severity::Warning
+                    && (category.contains(&info.category())
+                        || codes.contains(&(info.category(), info.code())))
+            }
+            WarningFilters::Empty => false,
+        }
+    }
+
+    pub fn union(&mut self, other: &WarningFilters) {
+        match (self, other) {
+            (s @ Self::Empty, _) => *s = other.clone(),
+            (_, Self::Empty) => (),
+            (Self::All, _) => (),
+            (s, Self::All) => *s = Self::All,
+            (
+                Self::Specified { category, codes },
+                Self::Specified {
+                    category: other_category,
+                    codes: other_codes,
+                },
+            ) => {
+                category.extend(other_category);
+                codes.extend(other_codes);
```

Comment:
A bit nitty but should this remove any codes from `codes` that have a category in `other_category`? i.e., if the whole category is filtered, then we don't need to keep any specific codes with that category during a union.